frontend/src/pages/warehouse/ModalAddWarehouse.js

```diff
@@ -0,0 +1,324 @@
+            }
+        }
+    };
+
+    const handleIdentifier = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            identifier: e.target.value
+        }));
+    };
+
+    const handleType = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            type: e
+        }));
+        setDropdownMenuName(e);
+    };
+
+    const handleCity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, city: e.target.value}
+        }));
+    };
+
+    const handleLineOne = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine1: e.target.value}
+        }));
+    };
+
+    const handleLineTwo = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine2: e.target.value}
+        }));
+    };
+
+    const handleTotalCapacity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            totalCapacity: e.target.value
+        }));
+    };
+
+    const addWarehouseHandler = (e) => {
+        e.preventDefault();
+        const dispatchersId = dispatchersList.map(dispatcher => dispatcher.id);
+        const updateWarehouseDto = {...warehouseDto, dispatchersId: dispatchersId, customerId: props.currentCustomerId}
+        let validationResult = validateWarehouse(warehouseDto, dispatchersId, dropdownMenuName);
+        setErrors(preState => ({
+            ...preState,
+            validationErrors: validationResult,
+            serverErrors: ''
+        }));
+        if (validationResult.length === 0) {
+            fetch('/customers/' + props.currentCustomerId + '/warehouses', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(updateWarehouseDto)
+            })
+                .then(function (response) {
+                        if (response.status !== 201) {
+                            setErrors({
+                                serverErrors: "Something go wrong, try later",
+                            });
+                        } else {
+                            setErrors(preState => ({
+                                ...preState,
+                                validationErrors: [],
+                                serverErrors: ''
+                            }));
+                            setDispatchersList([]);
+                            setDropdownMenuName("select type");
+                            props.onChange(false, warehouseDto)
+                        }
+                    }
+                )
+        }
+    };
+
+    const showDispatchers = dispatchersList.map(disp =>
+        <div key={disp.id}>
+            {disp.name + " " + disp.surname + ", username: " + disp.username}
```

Comment:
Use es6 string concatenation.

Reply:
got it

Reply:
between <div> scope I can't do something like `My name is ${person.name}`, but I can use 
<div> {disp.name} {disp.surname}, username: {disp.username} </div>
it's ok? I think, it seems clearer than it was before